pymove/visualization/folium.py

```diff
@@ -1468,3 +1436,754 @@ def plot_stops(
         base_map.save(outfile=filename)
     else:
         return base_map
+
+def _format_tags(line, slice_):
+    """
+    Create or format tags.
+
+    Parameters:
+    -----------
+    line: Line to add a tag.
+
+    slice_: Tag interval.
+    """
+    map_formated_tags = map(lambda tag: '{}: {}'.format(tag, line[tag]), slice_)
+
+    return '<br/>'.join(map_formated_tags)
+
+
+def _circle_maker(
+    iter_tuple, 
+    user_lat, 
+    user_lon, 
+    slice_tags, 
+    user_point, 
+    map_
+):
+    """
+    Return a circle.
+
+    Parameters:
+    -----------
+    iter_tuple: DataFrame iter_tuple.
+    user_lat: String.
+        Latitude column name.
+    user_lon: String.
+        Longitude column name.
+    slice_tags:
+
+    user_point: String.
+        Point color.
+    map_: Folium map.
+    """
+    _, line = iter_tuple
+
+    x = line[user_lat]
+    y = line[user_lon]
+
+    tags_formated = _format_tags(line, slice_tags)
+
+    folium.Circle(
+        radius=1,
+        location=[x, y],
+        popup=tags_formated,
+        color=user_point,
+        fill=False
+    ).add_to(map_)
+
+def plot_incial_end_points(
+    list_rowns, 
+    user_lat, 
+    user_lon, 
+    slice_tags, 
+    base_map,
+    map_
+):
+        """
+    Returns incial and end points.
+
+    Parameters:
+    -----------
+    list_rowns: List of DataFrame iter_tuple.
+    user_lat: String.
+        Latitude column name.
+    user_lon: String.
+        Longitude column name.
+    slice_tags:
+
+    user_point: String.
+        Point color.
+    map_: Folium map.
+    """
+    # plot the start tnz_point
+    line = list_rowns[0][1]
```

Comment:
Codacy found an issue: [unindent does not match any outer indentation level (F999)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5714604)